MathLib/mechanics_invariants.h

```diff
@@ -0,0 +1,189 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+#ifndef MATHLIB_MECHANICS_INVARIANTS_H
+#define MATHLIB_MECHANICS_INVARIANTS_H
+
+#include <Eigen/Dense>
+
+namespace MathLib
+{
+namespace Mechanics
+{
+/// Invariants used in mechanics, based on Kelvin representation of the vectors
+/// and matrices.
+/// The invariants are computed at process creation time.
+template <int KelvinVectorSize>
+struct Invariants final
+{
+    /// Kelvin mapping of deviatoric projection tensor. \f$A_{\rm dev} = P_{\rm
+    /// dev}:A\f$ for \f$A\f$ being a second order tensor.
+    static Eigen::Matrix<double, KelvinVectorSize, KelvinVectorSize> const
+        P_dev;
+    /// Kelvin mapping of spherical projection tensor. \f$A_{\rm sph} = P_{\rm
+    /// sph}:A\f$ for \f$A\f$ being a second order tensor.
+    static Eigen::Matrix<double, KelvinVectorSize, KelvinVectorSize> const
+        P_sph;
+    /// Kelvin mapping of 2nd order identity tensor.
+    static Eigen::Matrix<double, KelvinVectorSize, 1> const identity2;
+
+    /// Determinant of a matrix in Kelvin vector representation.
+    static double determinant(
```

Comment:
Yes together with trace and some others. Better explained in the [wiki](https://en.wikipedia.org/wiki/Invariants_of_tensors) for example.